packages/flutter_driver/lib/src/extension/extension.dart

```diff
@@ -98,19 +149,29 @@ typedef CommandHandlerCallback = Future<Result> Function(Command c);
 /// Signature for functions that deserialize a JSON map to a command object.
 typedef CommandDeserializerCallback = Command Function(Map<String, String> params);
 
-/// Signature for functions that run the given finder and return the [Element]
-/// found, if any, or null otherwise.
-typedef FinderConstructor = Finder Function(SerializableFinder finder);
+/// Used to expand the new Finder
+abstract class FinderExtension {
+
+  /// Identifies the type of finder to be used by the driver extension.
+  String get finderType;
+
+  /// Deserializes the finder from JSON generated by [SerializableFinder.serialize].
+  SerializableFinder deserialize(Map<String, String> params, DeserializeFinderFactory finderFactory);
+
+  /// Signature for functions that run the given finder and return the [Element]
+  /// found, if any, or null otherwise.
+  Finder createFinder(SerializableFinder finder);
+}
 
 /// The class that manages communication between a Flutter Driver test and the
 /// application being remote-controlled, on the application side.
 ///
 /// This is not normally used directly. It is instantiated automatically when
 /// calling [enableFlutterDriverExtension].
 @visibleForTesting
-class FlutterDriverExtension {
+class FlutterDriverExtension with DeserializeFinderFactory {
   /// Creates an object to manage a Flutter Driver connection.
-  FlutterDriverExtension(this._requestDataHandler, this._silenceErrors) {
+  FlutterDriverExtension(this._requestDataHandler, this._silenceErrors, List<FinderExtension> finders) {
```

Comment:
unfortunately adding a new named parameter here breaks some internal customers. Can we reland this with either an optional named parameter, or just a new named constructor for it?